src/charm.py

```diff
@@ -53,25 +61,14 @@ def __init__(self, *args):
             self._on_provide_certificate_action,
         )
 
-    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
-        """Triggered when the Juju config is changed.
+    def _on_install(self, event: ConfigChangedEvent) -> None:
```

Comment:
The type hint is wrong.
```suggestion
    def _on_install(self, event: InstallEvent) -> None:
```